docs/keyboard.rst

```diff
@@ -1,113 +1,187 @@
-   colors.
+Be careful to print it directly, though, our examples uses format string,
+``f'{ks!r}'`` for ``repr()``, because ``str(ks)`` may contain escape sequences
+beginning with (``KEY_ESCAPE``) and are generally unprintable.
 
-When the :attr:`~.Keystroke.is_sequence` property tests *True*, the value of
-:attr:`~.Keystroke.code` represents a unique application key of the keyboard.
+Special Keys
+~~~~~~~~~~~~
 
-:attr:`~.Keystroke.code` may then be compared with attributes of :class:`~.Terminal`,
-which are duplicated from those found in :linuxman:`curses(3)`, or those `constants
-<https://docs.python.org/3/library/curses.html#constants>`_ in :mod:`curses` beginning with phrase
-*KEY_*, as follows:
+The :attr:`~.Keystroke.is_sequence` property returns ``True`` for arrow keys,
+function keys, and any character key combined with modifiers (Ctrl, Alt, Shift).
 
-.. include:: all_the_keys.txt
+.. literalinclude:: ../bin/keyboard_special_keys.py
+   :language: python
+   :linenos:
 
-All such keystrokes can be decoded by blessed, there is a demonstration program, :ref:`keymatrix.py`
-that tests how many of them you can find !
+The ``str(key)`` value for :class:`~.Keystroke` should not be directly printed
+when :attr:`~.Keystroke.is_sequence` is True as done in this and other examples.
 
-delete
-------
+The :attr:`~.Keystroke.name` property provides a readable name for special keys,
+and can be used for basic equality tests, like in this "paint by arrow key" example:
 
-Typically, backspace is ``^H`` (8, or 0x08) and delete is ^? (127, or 0x7f).
+.. literalinclude:: ../bin/keyboard_arrow_paint.py
+   :language: python
+   :linenos:
 
-On some systems however, the key for backspace is actually labeled and transmitted as "delete",
-though its function in the operating system behaves just as backspace. Blessed usually returns
-"backspace" in most situations.
+Common key names include:
 
-It is highly recommend to accept **both** ``KEY_DELETE`` and ``KEY_BACKSPACE`` as having the same
-meaning except when implementing full screen editors, and provide a choice to enable the delete mode
-by configuration.
+* ``KEY_UP``, ``KEY_DOWN``, ``KEY_LEFT``, ``KEY_RIGHT`` - Arrow keys
+* ``KEY_ENTER`` - Enter/Return key
+* ``KEY_BACKSPACE``, ``KEY_DELETE`` - Backspace and Delete keys
+* ``KEY_TAB`` - Tab key
+* ``KEY_ESCAPE`` - Escape key
+* ``KEY_F1`` through ``KEY_F12`` - Function keys
+* ``KEY_PGUP``, ``KEY_PGDOWN`` - Page Up and Page Down
+* ``KEY_HOME``, ``KEY_END`` - Home and End keys
 
-Alt/meta
---------
+For regular characters without modifiers, :attr:`~.Keystroke.name` returns ``None``.
+
+Feel free to try the demonstration program, :ref:`keymatrix.py` to experiment
+with possible keyboard inputs and combinations.
+
+Modifiers
+~~~~~~~~~
+
+Standard keys with modifiers follow the pattern:
+
+* ``KEY_CTRL_A``
+* ``KEY_ALT_Q``
+* ``KEY_SHIFT_X``
+* ``KEY_CTRL_ALT_Y``
+* ``KEY_ALT_SHIFT_Z``
+* ``KEY_CTRL_ALT_SHIFT_L``
```

Comment:
Shift isn't reliable on my system.
KEY_SHIFT_X shows up as X
KEY_CTRL_ALT_SHIFT_L shows up as KEY_CTRL_ALT_L
KEY_ALT_SHIFT_Z works as expected

KEY_CTRL_BACKSPACE shows up as KEY_BACKSPACE, but CTRL does show up under extra

The rest work as expected

Reply:
Sorry this is mostly a documentation error, a few times I have supported shift with alt for alphanumerics, and I also tried case-sensative KEY_CTRL_ALT_z vs. KEY_CTRL_ALT_Z, but anyway, I have decided SHIFT is never in key_name, you can use ignore_case with magic method keystroke.key_ctrl_alt('z', 'ignore_case=False') if you really had to know the difference between upper and lower modified alphanumerics.

Also shift is never supported as distinguished with ctrl, control characters cannot be "shifted", they already are. maybe with kitty protocol, though.

Anyway I'll fix the documentaion and make note of SHIFT excluded from ``name`` when they are not application keys.